test/run-tests.py

```diff
@@ -615,28 +675,66 @@ def main(args):
     out_dir = tempfile.mkdtemp(prefix='sexpr-wasm-')
     out_dir_is_temp = True
 
+  variables['out_dir'] = os.path.abspath(out_dir)
+
+  allProcs = []
+  continuedErrors = 0
   try:
-    for i, p in enumerate(range(num_proc)):
-      args = (i, options, variables, inq, outq, out_dir)
-      proc = multiprocessing.Process(target=ProcessWorker, args=args)
-      proc.start()
+    if not options.no_multithreaded:
+      inq = multiprocessing.Queue()
+      outq = multiprocessing.Queue()
+      for info in infosToRun:
+        inq.put_nowait(info)
+      runWorkers = multiprocessing.Value('i', 1)
+      for i, p in enumerate(range(num_proc)):
+        args = (i, options, variables, inq, outq, out_dir, runWorkers)
+        proc = multiprocessing.Process(target=ProcessWorker, args=args)
+        allProcs.append(proc)
+        proc.start()
+      inq.close();
+
+      finished_tests = 0
+      while finished_tests < test_count:
+        try:
+          info, result = outq.get(True, 0.01)
+        except Queue.Empty:
+          status.UpdateTimer()
+          continue
 
-    finished_tests = 0
-    while finished_tests < test_count:
-      try:
-        info, result = outq.get(True, 0.01)
-      except Queue.Empty:
-        status.UpdateTimer()
-        continue
+        finished_tests += 1
+        HandleTestResult(status, info, result, options.rebase)
+        if options.stop and status.failed > continuedErrors:
+          shouldContinue = raw_input('Continue testing? (yes/no): ')
```

Comment:
Not a big fan of interactive scripts, but if you find it useful, OK. Would like this to be refactored into a `YesNo` function returning a bool though.